src/test/java/org/junitpioneer/jupiter/StdIoExtensionTests.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.engine.AbstractJupiterTestEngineTests;
+import org.junitpioneer.jupiter.StdIOExtension.StdIn;
+import org.junitpioneer.jupiter.StdIOExtension.StdOut;
+
+@ExtendWith(StdIOExtension.class)
+@DisplayName("StdIOExtension ")
+public class StdIoExtensionTests extends AbstractJupiterTestEngineTests {
+
+	final BasicCommandLineApp app = new BasicCommandLineApp();
+
+	@Nested
+	@DisplayName("with specific configuration ")
+	class ConfigurationTests {
+
+		@Test
+		@DisplayName("fails if parameter is not annotated with @StdIntercept")
+		void needsAnnotation() {
+			try {
+				executeTestsForMethod(StdIOExtensionConfigurationTests.class, "noAnnotation");
+			}
+			catch (ParameterResolutionException ex) {
+				assertThat(ex.getMessage()).startsWith("No ParameterResolver registered");
+			}
+		}
+
+		@Test
+		@DisplayName("fails if the parameter type is not StdIn or StdOut")
+		void needsType() {
+			try {
+				executeTestsForMethod(StdIOExtensionConfigurationTests.class, "badType");
+			}
+			catch (ParameterResolutionException ex) {
+				assertThat(ex.getMessage()).contains("Can only resolve parameter", "but was");
+			}
+		}
+
+		@Test
+		@DisplayName("resolves parameter for correct type and annotation")
+		void goodConfig() {
+			Assertions
+					.assertDoesNotThrow(
+						() -> executeTestsForMethod(StdIOExtensionConfigurationTests.class, "goodConfig"));
+		}
```

Comment:
*Minding liveticker ...the format looks very strange - do we have a general formatting problem here?...*

Reply:
This three tests **explicitly** check the annotation configuration. The first two are for no annotation and bad type - that's OK. But the last checks only for good configuration of `StdIn`. So there there is no **explicit** good configuration check for `StdOut` and therefore the related code is not covered by one of this configuration tests! 
As of now `StdOut` is only **indirectly** covered by the later two for checking the captured lines. If you only want to check **indirectly** that you can remove the test of `StdIn` as it's checked by the later tests.

I prefer to split the tests for checking the configuration and the output, so I suggest to add one for `StdOut` too.